mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -872,6 +872,10 @@ def method_call(self, method_name: str, args: T.List[TYPE_var], kwargs: TYPE_kwa
             args = flatten(args)
         if not getattr(method, 'no-second-level-holder-flattening', False):
             args, kwargs = resolve_second_level_holders(args, kwargs)
+        if len(self.interpreter.active_projectname) == 0:
```

Comment:
```suggestion
        if not self.interpreter.active_projectname:
```

Reply:
I had tried this initially, [but `active_projectname` is initialized to `''`](https://github.com/mesonbuild/meson/blob/f83dca32fe340727ada08b46d70bfc6dbba52d84/mesonbuild/interpreter/interpreter.py#L277)

Reply:
Actually, I was incorrect: I must've previously tried something that I thought was equivalent like
```py
if self.interpreter.active_projectname is not None:
```
but it turns out that this isn't equivalent to 
```py
if not self.interpreter.active_projectname:
```

Thanks for bringing this to my attention. I admit that I've spent a lot more time writing C than writing Python so my Python knowledge isn't particularly comprehensive.